csvcubed/csvcubed/cli/build.py

```diff
@@ -68,14 +68,17 @@ def build(
             ]
 
             with open(validation_errors_file_out, "w+") as f:
-                json.dump(all_errors, f, indent=4)
+                json.dump(all_errors, f, indent=4, default=serialize_sets)
+            # with open(output_directory / validation_errors_file_out, "w+") as f:
+            #     json.dump(all_errors, f, indent=4, default=serialize_sets)
 
         if fail_when_validation_error_occurs and len(validation_errors) > 0:
             exit(1)
+    else:
```

Comment:
When a cube is built but validation_errors have been returned, its not possible to write the cube using QbWriter as exceptions are raised. This change was to avoid this happening.  It would be better to wrap this in a try-except block or to update QbWriter to better handle known exception cases.  
e.g.
self = QbWriter(cube=Cube(metadata=CatalogMetadata(title='Tests/test-cases/config/schema-cube-data-config-ok', identifier='sc...be=Cube(metadata=CatalogMetadata(title='Tests/test-cases/config/schema-cube-data-config-ok', identifier='schema-id'))))
t = <class 'csvcubed.models.cube.qb.components.measuresdimension.QbMultiMeasureDimension'>

    def _get_single_column_of_type(
        self, t: Type[QbColumnarDsdType]
    ) -> QbColumn[QbColumnarDsdType]:
        cols = get_columns_of_dsd_type(self.cube, t)
        if len(cols) != 1:
>           raise ValueError(
                f"Found {len(cols)} columns with component type {t} in cube. Expected 1."
            )
E           ValueError: Found 0 columns with component type <class 'csvcubed.models.cube.qb.components.measuresdimension.QbMultiMeasureDimension'> in cube. Expected 1.

../../csvcubed/writers/qbwriter.py:943: ValueError